src/validateRegisterForm.test.js

```diff
@@ -12,19 +16,67 @@ describe(`Function 'validateRegisterForm':`, () => {
       .toBe('object');
   });
 
-  it(`should return success message for the valid input`, () => {
-    const isValid = validateRegisterForm('test@mail.com', 'P@ssword1!');
+  it(`should return success message for valid email and password`, () => {
+    const returnValue = getReturnValue(200, 'Email and password are valid.');
+
+    expect(validateRegisterForm('test@mail.com', 'P@ssword1!'))
+      .toStrictEqual(returnValue);
+
+    expect(validateRegisterForm('user123@mail.com', 'MeAw@som2P0ss!'))
+      .toStrictEqual(returnValue);
+
+    expect(validateRegisterForm('stm@test.ua!!!', 'P@ssword1!'))
+      .toStrictEqual(returnValue);
+  });
+
+  it(`should return error for valid email and invalid password`, () => {
+    const returnValue = getReturnValue(422, 'Password is invalid.');
+
+    expect(validateRegisterForm('test@mail.com', 'P@ssword'))
+      .toStrictEqual(returnValue);
+
+    expect(validateRegisterForm('asura@scans.uk.1222$$$$', 'Password'))
+      .toStrictEqual(returnValue);
+
+    expect(validateRegisterForm('animal@control.pu', '123@@@p'))
+      .toStrictEqual(returnValue);
+  });
+
+  it(`should return error for valid password and invalid email`, () => {
+    const returnValue = getReturnValue(422, 'Email is invalid.');
+
+    expect(validateRegisterForm('test!mail.com', 'P@ssword1'))
+      .toStrictEqual(returnValue);
+
+    expect(validateRegisterForm('111!!sasd@mail.com', 'S0m!P@ss123'))
+      .toStrictEqual(returnValue);
 
-    expect(isValid.code).toBe(200);
-    expect(isValid.message).toBe('Email and password are valid.');
+    expect(validateRegisterForm('connings@burg!.com', 'P@ssw))rd1'))
+      .toStrictEqual(returnValue);
+
+    expect(validateRegisterForm('.non@binary!.com.', 'P@wed12da'))
+      .toStrictEqual(returnValue);
```

Comment:
The email '.non@binary!.com.' contains invalid characters and structure. Ensure that the `validateRegisterForm` function is expected to handle such cases as invalid.